src/Mages.Core/Runtime/Functions/StandardFunctions.cs

```diff
@@ -659,6 +659,7 @@ public static class StandardFunctions
         {
             return Curry.MinTwo(Where, args) ??
                 If.Is<Function, Double[,]>(args, (f, m) => m.Where(f)) ??
+                If.Is<Function, IEnumerable>(args, (f, enu) => enu.Where(f)) ??

```

Comment:
This one will always leave out the dictionary, as the dictionary implements an `IEnumerable`.